app/javascript/hyacinth_ui_v1/components/projects/permissions/PermissionsEditor.js

```diff
@@ -0,0 +1,300 @@
+          { /* Note 2: We're Using a label for a wider clickable checkbox area. */ }
+          <label
+            id={`label-${projectPermission.user.id}-${action}`}
+            htmlFor={`checkbox-${projectPermission.user.id}-${action}`}
+            className="w-100"
+            >
+            <Form.Check
+              id={`checkbox-${projectPermission.user.id}-${action}`}
+              label="&nbsp;"
+              disabled={disabledCheckbox}
+              type="checkbox"
+              checked={projectPermission.actions.includes(action)}
+              onChange={(e) => {
+                if (readOnly) return;
+                updatePermissionsData(projectPermission.user.id, action, e.target.checked);
+              }}
+            />
+          </label>
+          { /* eslint-enable jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for */ }
+        </td>
+      );
+    });
+  };
+
+  const renderProjectPermissions = () => {
+    return projectPermissions.map(projectPermission => (
+      <tr key={projectPermission.user.id}>
+        <td key="name">
+          {projectPermission.user.fullName}
+        </td>
+
+        {renderProjectPermission(projectPermission)}
+
+        { !readOnly && (
+          <td key="remove" className="text-center">
+            <RemoveButton onClick={() => {removeUser(projectPermission.user.id); }} />
+          </td>
+        ) }
+      </tr>
+    ));
+  };
+
+  const addSelectedUser = () => {
+    if (currentAddUserSelection == null) return;
+    setPermissionsChanged(true);
+
+    const { projectPermissionActions: { readObjectsAction } } = actionsData;
+    const userId = currentAddUserSelection.value;
+    const user = allUsers.find(u => u.id === userId);
+
+    setProjectPermissions(produce(projectPermissions, (draft) => {
+      draft.push({
+        user,
+        project: {
+          stringKey: projectStringKey,
+        },
+        actions: [readObjectsAction],
+      });
+    }));
+
+    // If the added user was previously among the set of removedUserIds,
+    // then remove the added user's id from removedUserIds.
+    if (removedUserIds.has(userId)) {
+      setRemovedUserIds(
+        produce(removedUserIds, (draft) => {
+          draft.delete(userId);
+        }),
+      );
+    }
+
+    setCurrentAddUserSelection(null); // this will clear out the current select item
+  };
+
+  const renderUserAdd = () => {
+    const addedUserIds = projectPermissions.map(permission => permission.user.id);
+    const nonAddedUsers = allUsers.filter(user => !addedUserIds.includes(user.id));
+    return (
+      <tr key="user-add">
+        <td colSpan={actionsData.projectPermissionActions.actions.length + 1}>
+          <Select
+            placeholder="Add a user..."
+            value={currentAddUserSelection}
+            options={nonAddedUsers.map(user => ({ value: user.id, label: user.fullName }))}
+            onChange={(val) => { setCurrentAddUserSelection(val); }}
```

Comment:
```suggestion
            onChange={(val) => setCurrentAddUserSelection(val) }
```

Reply:
I think it needs to be `onChange={val => setCurrentAddUserSelection(val)}`, but I see what you mean.  Will change, thanks!